jobs/src/main/scala/dcos/metronome/model/JobRunSpec.scala

```diff
@@ -55,13 +55,22 @@ object JobRunSpec {
           violations += RuleViolation(jobRunSpec, errorMessage)
         }
       }
-      val envVarDefinedSecretNames = jobRunSpec.env.values.collect { case EnvVarSecret(secretName) => secretName }.toSet
+      val definedSecretNames =
+        jobRunSpec.env.values.collect { case EnvVarSecret(secretName) => secretName }.toSet ++
+          jobRunSpec.volumes.collect { case SecretVolume(_, secret) => secret }.toSet
       val providedSecretNames = jobRunSpec.secrets.keySet
 
       check(jobRunSpec.cmd.isDefined || jobRunSpec.docker.exists(d => d.image.nonEmpty) || jobRunSpec.ucr.nonEmpty, JobRunSpecMessages.cmdOrDockerValidation)
-      check(envVarDefinedSecretNames == providedSecretNames, JobRunSpecMessages.secretsValidation(envVarDefinedSecretNames, providedSecretNames))
+      check(definedSecretNames == providedSecretNames, JobRunSpecMessages.secretsValidation(definedSecretNames, providedSecretNames))
```

Comment:
I'm not sure that the `definedSecretNames` needs to == `definedSecretNames`.   What is required is that a `definedSecretNames` is in the set of providedSecretNames.
granted... I don't have a use case for providing a secret which isn't used... however I could see someone debugging / troubleshooting and removing the use of the secret and being frustrated with needing to also remove the provided secret name as well.    
This just seems to be more strict than necessary

Reply:
Queue unused imports discussion 😆 🤣 

There's utility in it being strict and in this case I don't see much drawback to being strict. I'd vote to keep the strictness.

(And, also, I'd vote against making it more strict than it was, but that doesn't seem to be the case here)